packages/frontend/web-ui/src/auth/hooks/useRegisterConfirm.spec.js

```diff
@@ -0,0 +1,281 @@
+        expect(useSelector).toHaveBeenCalled();
+      });
+
+      it('should return a rejected status', () => {
+        expect(status).toBe(STATUS_REJECTED);
+      });
+
+      it('should return an error message Unexpected error', () => {
+        expect(errorMessage).toBe(UNEXPECTED_ERROR_MESSAGE);
+      });
+
+      afterAll(() => {
+        jest.resetAllMocks();
+        jest.clearAllMocks();
+      });
+    });
+
+    describe('when called, and token is null', () => {
+      let result;
+      let status;
+      let errorMessage;
+
+      beforeAll(async () => {
+        tokenFixture = null;
+        authErrorMessageFixture = UNEXPECTED_ERROR_MESSAGE;
+
+        useDispatch.mockReturnValue(dispatchMock);
+        useSelector.mockImplementation(() => ({
+          token: tokenFixture,
+          errorMessage: authErrorMessageFixture,
+        }));
+
+        await act(() => {
+          result = renderHook(() => useRegisterConfirm()).result;
+        });
+        status = result.current.status;
+        errorMessage = result.current.errorMessage;
+      });
+
+      it('should called useDispatch()', () => {
+        expect(useDispatch).toHaveBeenCalled();
+      });
+
+      it('should called useSelector() ', () => {
+        expect(useSelector).toHaveBeenCalled();
+      });
+
+      it('should return a rejected status', () => {
+        expect(status).toBe(STATUS_REJECTED);
+      });
+
+      it('should return an error message Unexpected error', () => {
+        expect(errorMessage).toBe(UNEXPECTED_ERROR_MESSAGE);
+      });
+
+      afterAll(() => {
+        jest.resetAllMocks();
+        jest.clearAllMocks();
+      });
+    });
+
+    afterAll(() => {
+      Object.defineProperty(window, 'location', {
+        value: previousLocation,
+        configurable: true,
+      });
+    });
+  });
+
+  describe('having a window with location.href without code query', () => {
+    let previousLocation;
+    let locationFixture;
+
+    beforeAll(() => {
+      previousLocation = window.location;
+      locationFixture = new URL('http://corniegame.com/auth/path?code=');
+
+      Object.defineProperty(window, 'location', {
+        value: new URL(locationFixture),
+        configurable: true,
+      });
+    });
+
+    describe('when called, and code query not exists and the error grid is showed', () => {
+      let result;
+      let status;
+      let errorMessage;
+
+      beforeAll(async () => {
+        useDispatch.mockReturnValue(dispatchMock);
+        useSelector.mockImplementation(() => ({
+          token: tokenFixture,
+          errorMessage: authErrorMessageFixture,
+        }));
+
+        await act(() => {
+          result = renderHook(() => useRegisterConfirm()).result;
+        });
+
+        status = result.current.status;
+        errorMessage = result.current.errorMessage;
+      });
+
+      it('should called useDispatch()', () => {
+        expect(useDispatch).toHaveBeenCalled();
```

Comment:
`toHaveBeenCalledTimes` is more accurate.

In addition, assert mock arguments with `toHaveBeenCalledWith`.

Reply:
With `toHaveBeenCalledTimes` was called three times because the component does 3 iterations until it is rendered. I have done the `toHaveBeenCalledWith`.